### DISCUSSION TITLE
serialise all subclass properties

### DISCUSSION THREAD

**Discussion by mjmdavis** [pydantic/pydantic#9879] [Question] [answered]
How should one describe that a property is an instance of a subclass of a given class?

```python
from pydantic import BaseModel
from typing import Optional

class B(BaseModel):
    carrot: Optional[str] = None

class C(B):
    banana: str = 'banana'

class Container(BaseModel):
    c:B

container = Container(c=C())
container.model_dump_json()
```
gives `'{"c":{"carrot":null}}'`

I would expect it to give `'{"c":{"carrot":null,"banana":"banana"}}'`

How can I properly describe this to get the result I expect?

**Comment by uriyyo** [comment, marked as answer]
Hi @mjmdavis,

You can use `SerializeAsAny`:

```py
from pydantic import BaseModel, SerializeAsAny
from typing import Optional


class B(BaseModel):
    carrot: Optional[str] = None


class C(B):
    banana: str = 'banana'


class Container(BaseModel):
    c: SerializeAsAny[B]


container = Container(c=C())
print(container.model_dump_json())
```

More information here - https://docs.pydantic.dev/latest/concepts/serialization/#serializing-with-duck-typing

**Reply by mjmdavis** [reply]
Amazing! Thank you. This is exactly what I was looking for.
